fingerprint_verification.py

```diff
@@ -6,39 +6,103 @@
 
 def read_print(filename):
     """Reads the given print and returns a dictionary with all info."""
-    # Your turn!
-
-
-
-
-
-
-# You can add your various check functions below
-
-
-
-
-
-
-
-
-
-
-
-
-
-
-
-
-
-
-
-
+    fingerprint_data = {}
+    with open(filename) as fh:
+        name = fh.readline().strip()
+        width = int(fh.readline().strip())
+        height = int(fh.readline().strip())
+        
+        fingerprint_data['name'] = name
+        fingerprint_data['width'] = width
+        fingerprint_data['height'] = height
+
+        fingerprint = [] 
+
+        for line in filename:
+            cleaned_line = line.rstrip()  # Keep trailing spaces but remove newlines
+        # If the line is empty, create a row of spaces with the correct width
+        if not cleaned_line:
+            cleaned_line = ' ' * width
+        fingerprint.append(cleaned_line)
+
+        if not fingerprint:  # Check if the fingerprint list is empty
+            print(f"Warning: No fingerprint data found in {filename}")
+        else:
+            fingerprint_data['fingerprint'] = fingerprint
+
+    print(fingerprint_data)
+    return fingerprint_data
+
+
+def simple_check(fingerprint_data1, fingerprint_data2):
+    """
+    Compares the fingerprint portions of two fingerprint dictionaries.
+    Returns True if the fingerprints are the same, otherwise False.
+    """
+    return fingerprint_data1['fingerprint'] == fingerprint_data2['fingerprint']
+
+def variant_check(fingerprint_data1, fingerprint_data2):
+    """
+    Compares the fingerprint portions of two fingerprint dictionaries
+    and calculates the percentage of matching pixels.
+    Returns True if the matching percentage is 95% or higher, otherwise False.
+    """
+    fingerprint1 = fingerprint_data1['fingerprint']
+    fingerprint2 = fingerprint_data2['fingerprint']
+
+    matching_pixels = 0
+    total_pixels = 0
+
+    for i in range(len(fingerprint1)):
+        for j in range(len(fingerprint1[i])):
+            total_pixels += 1
+            if fingerprint1[i][j] == fingerprint2[i][j]:
+                matching_pixels += 1
```

Comment:
This is good, it just breaks with your read data because of the differences in row lengths owing to the `rstrip`